components/homme/src/share/gllfvremap_mod.F90

```diff
@@ -0,0 +1,1707 @@
+     module procedure gfr_dyn_to_fv_phys_dom_mt
+  end interface gfr_dyn_to_fv_phys
+
+  interface gfr_fv_phys_to_dyn
+     module procedure gfr_fv_phys_to_dyn_hybrid
+     module procedure gfr_fv_phys_to_dyn_dom_mt
+  end interface gfr_fv_phys_to_dyn
+
+  interface gfr_fv_phys_to_dyn_topo
+     module procedure gfr_fv_phys_to_dyn_topo_hybrid
+     module procedure gfr_fv_phys_to_dyn_topo_dom_mt
+  end interface gfr_fv_phys_to_dyn_topo
+
+contains
+
+  subroutine gfr_init(par, elem, nphys, check)
+    use kinds, only: iulog
+    use dimensions_mod, only: nlev
+    use parallel_mod, only: parallel_t, abortmp
+    use quadrature_mod, only : gausslobatto, quadrature_t
+    use control_mod, only: ftype
+
+    type (parallel_t), intent(in) :: par
+    type (element_t), intent(in) :: elem(:)
+    integer, intent(in) :: nphys
+    logical, intent(in), optional :: check
+
+    real(real_kind) :: R(npsq,nphys_max*nphys_max), tau(npsq)
+    integer :: nphys2
+
+    gfr%check = .false.
+    if (present(check)) gfr%check = check
+
+    gfr%tolfac = one
+    if (par%masterproc) &
+         write(iulog,  '(a,i3,a,l2,a,i2)') 'gfr> init nphys', nphys, ' check', gfr%check, ' ftype', ftype
+
+    if (nphys > np) then
+       ! The FV -> GLL map is defined only if nphys <= np. If we ever are
+       ! interested in the case of nphys > np, we will need to write a different
+       ! map. See "!assume" annotations for mathematical assumptions in
+       ! particular routines.
+       call abortmp('gllfvremap_mod: nphys must be <= np')
+    end if
+    if (qsize == 0) then
+       call abortmp('gllfvremap_mod: qsize must be >= 1')
+    end if
+
+    gfr%have_fv_topo_file_phis = .false.
+    gfr%nphys = nphys
+    gfr%npi = max(2, nphys)
+    nphys2 = nphys*nphys
+
+    call gfr_init_w_gg(np, gfr%w_gg)
+    call gfr_init_w_gg(gfr%npi, gfr%w_sgsg)
+    call gfr_init_w_ff(nphys, gfr%w_ff)
+    call gfr_init_M_gf(np, nphys, gfr%M_gf, .true.)
+    gfr%g2f_remapd(:,:,:nphys2) = reshape(gfr%M_gf(:,:,:nphys,:nphys), (/np,np,nphys2/))
+    call gfr_init_M_gf(gfr%npi, nphys, gfr%M_sgf, .false.)
+    call gfr_init_R(gfr%npi, nphys, gfr%w_sgsg, gfr%M_sgf, R, tau)
+    call gfr_init_interp_matrix(gfr%npi, gfr%interp)
+    call gfr_init_f2g_remapd(gfr, R, tau)
+
+    allocate(gfr%fv_metdet(nphys2,nelemd), &
+         gfr%D_f(nphys,nphys,2,2,nelemd), gfr%Dinv_f(nphys,nphys,2,2,nelemd), &
+         gfr%qmin(nlev,max(1,qsize),nelemd), gfr%qmax(nlev,max(1,qsize),nelemd), &
+         gfr%phis(nphys2,nelemd), gfr%spherep_f(nphys,nphys,nelemd))
+    call gfr_init_fv_metdet(elem, gfr)
+    call gfr_init_geometry(elem, gfr)
+  end subroutine gfr_init
+
+  subroutine gfr_finish()
+    if (.not. allocated(gfr%fv_metdet)) return
+    deallocate(gfr%fv_metdet, gfr%D_f, gfr%Dinv_f, gfr%qmin, gfr%qmax, gfr%phis, &
+         gfr%spherep_f)
+  end subroutine gfr_finish
+
+  subroutine gfr_dyn_to_fv_phys_hybrid(hybrid, nt, hvcoord, elem, nets, nete, &
```

Comment:
Personally I'm not a fan of the "self-documenting code" idea, so I would appreciate some descriptions for each of these sub-routines. I also think comment lines of 60-100 dashes between subroutines make it a lot easier to scan big modules like this. So you can leave it this way if you like, but I think it's always good practice to think about the poor postdoc who's gonna have to try to trace through the code in 5-10 years.

Reply:
I'll add more documentation and delimit groups of related functions.

Reply:
I pushed a commit that acts on all PR review comments up to this point.